app/models/user.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  include Devise::JWT::RevocationStrategies::JTIMatcher
+
+  default_scope { where(active: true) }
+
+  devise :two_factor_authenticatable,
+         :registerable,
+         :validatable,
+         :confirmable,
+         :jwt_authenticatable,
+         :two_factor_backupable,
+         :recoverable,
+         jwt_revocation_strategy: self,
+         otp_secret_encryption_key: Settings.otp_secret_encryption_key
+
+  has_many :access_grants,
+           class_name: 'Doorkeeper::AccessGrant',
+           foreign_key: :resource_owner_id,
+           dependent: :delete_all
+
+  has_many :access_tokens,
+           class_name: 'Doorkeeper::AccessToken',
+           foreign_key: :resource_owner_id,
+           dependent: :delete_all
+
+  validate :password_complexity
+
+  has_many :server_provider_permissions
+  has_many :server_providers, through: :server_provider_permissions
+
+  has_many :repository_permissions
+  has_many :repositories, through: :repository_permissions
+
+  def server_provider_permission(server_provider_id)
+    server_provider_permissions.find_by(server_provider_id: server_provider_id)
+  end
+
+  def set_server_provider_permission(server_provider_id, permission)
+    perm = server_provider_permissions.find_or_initialize_by(server_provider_id: server_provider_id)
+    perm.permission = permission
+    perm.save
+  end
+
+  def repository_permission(repository_id)
+    repository_permissions.find_by(repository_id: repository_id)
+  end
+
+  def mark_as_deleted
+    update_columns(email: "deleted_email_#{Kernel.rand(1_000_000_000)}", name: nil, active: false)
```

Comment:
@speedywizard  Just found that the email pattern is not a valid pattern so it fails when users want to delete their account. 
This email pattern gives something like "deleted_email_662959238" so I thing it should be something like
`"deleted_email_#{Kernel.rand(1_000_000_000)}@example.com"` so it don't fail.

Reply:
@travis-architect , AFAIK, `update_columns` doesn't run any validations. Are you sure that it's failing because of this?

Reply:
@speedywizard I did the same in console and it was failing there too, so when I added the "@example.com" part, it got updated.

Reply:
@travis-architect  ,hm, sorry, my bad - I just checked doc. It's pretty strange to me that Rails behaves in such an inconsistent way. `update_column` indeed skips validations, but `update_columns` does not